tests/core/test_processor.py

```diff
@@ -245,11 +243,93 @@
 
     tracker_0 = default_processor.tracker_store.retrieve(sender_ids[0])
```

Comment:
do we really need the `await asyncio.sleep(5) ` part?

Reply:
This is the original test. If I remove the `sleep(5)` the test fails. But I agree that this is not nice. Is there a way of, e.g., asking in an infinite loop if things are ready?

Reply:
you could check if the `scheduler` offers a function to check if all jobs were run or you probably could also check for the length of the jobs in the scheduler in a loop